### PR TITLE
Added missing protection during liveliness assertion

### PR DESCRIPTION
**Pull request by jsantiago-eProsima**
Signed-off-by: Javier Santiago <javiersantiago@eprosima.com>

## Description

Added missing protection during liveliness assertion. This may cause erroneous memory access in some cases.

## Contributor Checklist
- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
- [ ] Any new/modified methods have been properly documented using Doxygen. 
- [ ] Fast DDS test suite has been run locally. 
- [X] Changes are ABI compatible. 
- [X] Changes are API compatible. 
- [ ] Documentation builds and tests pass locally. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Review by MiguelBarro — DISMISSED** [review]
LGTM

File: src/cpp/rtps/writer/LivelinessManager.cpp [hunk 1]
```diff
@@ -218,6 +218,8 @@ bool LivelinessManager::assert_liveliness(
         return false;
     }
 
+    std::unique_lock<std::mutex> lock(mutex_);
```
Comment (MiguelCompany) [inline src/cpp/rtps/writer/LivelinessManager.cpp:221]:
```suggestion
    std::lock_guard<std::mutex> lock(mutex_);
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Comment by MiguelCompany** [comment]
@jsantiago-eProsima I think we can close this since #2801 has already been merged into master

**Comment by jsantiago-eProsima** [comment]
Agreed. Closing.

**Pull request closed without merging**